### PR TITLE
Java support, create module, package and deploy lambda

### PR DESCRIPTION
**Pull request by doapp-jeremy**

### CODE REVIEW

**Comment by sixlettervariables** [comment]
The stylistic changes make it harder to review the core deltas.

File: lib/commands/module_create.js [hunk 1]
```diff
@@ -271,6 +360,7 @@ CMD.prototype._createSkeleton = Promise.method(function() {
   actionTemplateJson.apiGateway.cloudFormation.Path = _this._module.name + '/' + _this._module.action;
   actionTemplateJson.apiGateway.cloudFormation.Method = 'GET';
   actionTemplateJson.apiGateway.cloudFormation.Type = 'AWS';
+  actionTemplateJson.lambda.cloudFormation.Runtime = _this._module.runtime;
```
Comment (sixlettervariables) [inline lib/commands/module_create.js]: Probably belongs in the if-protected block just before the switch block below (line 366), as that detects if the module is a lambda method.

**Comment by ac360** [comment]
Doesn't matter what it looks like.  This is a start for Java support! :+1: :100:

**Comment by doapp-jeremy** [comment]
Yeah, some of the files hand't been formatted with jscs yet.  Should I have created a separate pull-request with just those changes first?

File: lib/commands/module_create.js (hunk 1 above)
Comment (doapp-jeremy) [inline lib/commands/module_create.js]: Good catch, this is my first pull-request, can/should I just amend it?

**Comment by Ninir** [comment]
@doapp-jeremy It would be easier to review I guess. Nice work! :+1:

**Comment by yaochen** [comment]
this is awesome!

any eta on when itll be merge/part of a release?

**Comment by fdragu** [comment]
I'm interested in the progress on this, as my teams will mainly use Java based lambdas. However, we are using gradle as the build tool rather than maven - any chance to make it so I can plug in my own gradle based package manager?

**Comment by jmpage** [comment]
@ac360 It looks like the code has changed pretty significantly since this was written. I think it might make sense to close this PR for now and as part of the java support roadmap item, take care to create a good abstraction around builds so that other build tools like gradle or sbt can be easily used in the future.

**Comment by eahefnawy** [comment]
Closing this PR as it's severely outdated. Thanks tons @doapp-jeremy for your work on this :blush:

**Pull request closed without merging**